src/include/cephfs/libcephfs.h

```diff
@@ -1572,6 +1572,48 @@ int ceph_ll_getlk(struct ceph_mount_info *cmount,
 int ceph_ll_setlk(struct ceph_mount_info *cmount,
 		  Fh *fh, struct flock *fl, uint64_t owner, int sleep);
 
+/*
+ * Delegation support
+ *
+ * Delegations are way for an application to request exclusive or
+ * semi-exclusive access to an Inode. The client requests the delegation and
+ * if it's successful it can reliably cache file data and metadata until the
+ * delegation is recalled.
+ *
+ * Recalls are issued via a callback function, provided by the application.
+ * Callback functions should act something like signal handlers.  You want to
+ * do as little as possible in the callback. Any major work should be deferred
+ * in some fashion as it's difficult to predict the context in which this
+ * function will be called.
+ *
+ * Once the delegation has been recalled, the application should return it as
+ * soon as possible. The application has client_deleg_timeout seconds to
+ * return it, after which the cmount structure is forcibly unmounted and
+ * further calls into it fail.
+ *
+ * The application can frob the client_deleg_timeout config option to suit its
```

Comment:
Yeah, basically it can be set however the application likes, and yes, we probably need some guidelines on how large it can be set without the client ending up evicted.

Reply:
BTW: do you have suggestions for how we should phrase that limit? I just left it vague:

"it should take care to choose a value that allows it to avoid forcible eviction..."

A documented and concrete limit would be better, but I'm not sure what formula to give.

Reply:
Well, the client timeouts are documented somewhere, right? So probably it's best to do it in terms of that. You could even programmataically enforce it!
(Yeah, you should do that. Safety first!)